plugins/automation/app/models/discourse_automation/stat.rb

```diff
@@ -0,0 +1,128 @@
+# frozen_string_literal: true
+#
+module DiscourseAutomation
+  class Stat < ActiveRecord::Base
+    self.table_name = "discourse_automation_stats"
+
+    def self.log(automation_id, run_time = nil)
+      if block_given? && run_time.nil?
+        start_time = Process.clock_gettime(Process::CLOCK_MONOTONIC)
+        result = yield
+        run_time = Process.clock_gettime(Process::CLOCK_MONOTONIC) - start_time
+        result
+      end
+    ensure
+      update_stats(automation_id, run_time)
+    end
+
+    def self.fetch_period_summaries
+      today = Date.current
+
+      # Define our time periods
+      periods = {
+        last_day: {
+          start_date: today - 1.day,
+          end_date: today,
+        },
+        last_week: {
+          start_date: today - 7.days,
+          end_date: today,
+        },
+        last_month: {
+          start_date: today - 30.days,
```

Comment:
On 1st March this will evaluate to 30th Jan:

```ruby
Date.new(2025, 3, 1) - 30.days
#=> Thu, 30 Jan 2025
```

Might want to use `- 1.month` instead:

```ruby
Date.new(2025, 3, 1) - 1.month
=> Sat, 01 Feb 2025
```

Reply:
I fixed it

Reply:
thanks @jjaffeux